### PR TITLE
Encryption enhancements

### PR DESCRIPTION
**Pull request by roysjosh**
Fixes for #176

### CODE REVIEW

File: op25/gr-op25_repeater/lib/p25_crypt_algs.cc [hunk 1]
```diff
@@ -110,39 +110,48 @@ bool p25_crypt_algs::process(packed_codeword& PCW) {
...
             break;
         default:
             rc = false;
             break;
     }
-    if ((d_fr_type == FT_LDU1) || (d_fr_type == FT_LDU2)) {
+    if (d_pr_type == PT_P25_PHASE1) {
         //FDMA
         offset += (d_adp_position * 11) + 267 + ((d_adp_position < 8) ? 0 : 2); // voice only; skip LCW and LSD
         d_adp_position = (d_adp_position + 1) % 9;
```
Comment (roysjosh) [inline op25/gr-op25_repeater/lib/p25_crypt_algs.cc:149]: I could work on this section a bit more & eliminate `d_adp_position`

File: op25/gr-op25_repeater/lib/p25p2_tdma.cc [hunk 2]
```diff
@@ -606,23 +628,42 @@ int p25p2_tdma::handle_packet(uint8_t dibits[], const uint64_t fs)
...
+			d_tdma_slot_for_2v = current_slot;
+
+		if (current_slot <= (int) d_tdma_slot_for_2v)
+			current_slot += 5;
+
+		current_slot -= (d_tdma_slot_for_2v + 1);
+
+		if (current_slot != burst_id && current_slot > burst_id) {
+			int need_to_skip = current_slot - burst_id;
+			// XXX determine if the 2V frame was missed?
```
Comment (roysjosh) [inline op25/gr-op25_repeater/lib/p25p2_tdma.cc:645]: As the comment says, can we use something else to determine other losses? Maybe the I-ISCH info...

File: op25/gr-op25_repeater/lib/p25p2_tdma.cc [hunk 3]
```diff
@@ -606,23 +628,42 @@ int p25p2_tdma::handle_packet(uint8_t dibits[], const uint64_t fs)
...
+			fprintf(stderr, "%i voice frame(s) missing; expecting %uV_%u but got %uV_%u\n", need_to_skip, (burst_id == 4 ? 2 : 4), burst_id, (current_slot == 4 ? 2 : 4), current_slot);
+			//crypt_algs.skip(need_to_skip * 4);
```
Comment (roysjosh) [inline op25/gr-op25_repeater/lib/p25p2_tdma.cc:647]: Oops, this can be deleted

**Comment by boatbod** [comment]
My op25 gnuradio version independent feature enhancements are typically a first written against "dev" branch, merged up to main and then merged down to gr-specific branches.  i.e. dev->main->gr310.  Would it be possible for you to resubmit this PR with dev as the target branch?  Thanks.

**Comment by roysjosh** [comment]
> My op25 gnuradio version independent feature enhancements are typically a first written against "dev" branch, merged up to main and then merged down to gr-specific branches. i.e. dev->main->gr310. Would it be possible for you to resubmit this PR with dev as the target branch? Thanks.

Rebased to dev

**Comment by boatbod** [comment]
Merged to dev.  As soon as I get a chance to run some basic regression I'll promote it forward.

**Pull request merged**